addons/mail_group/models/mail_group_message.py

```diff
@@ -0,0 +1,271 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+import json
+
+from werkzeug import urls
+
+from odoo import _, api, fields, models, tools
+from odoo.addons.http_routing.models.ir_http import slug
+from odoo.exceptions import UserError
+from odoo.osv import expression
+from odoo.tools import email_normalize
+
+_logger = logging.getLogger(__name__)
+
+
+class MailGroupMessage(models.Model):
+    """Incoming email that will be posted on the <mail.group>."""
+
+    _name = 'mail.group.message'
+    _description = 'Mailing List Message'
+    _rec_name = 'subject'
+    _order = 'id DESC'
+
+    # <mail.message> fields, can not be done with inherits because it will impact
+    # the performance of the <mail.message> model (different cache, so the ORM will need
+    # to do one more SQL query to be able to update the <mail.group.message> cache)
+    attachment_ids = fields.Many2many(related='mail_message_id.attachment_ids', readonly=False)
+    author_id = fields.Many2one(related='mail_message_id.author_id', readonly=False)
+    email_from = fields.Char(related='mail_message_id.email_from', readonly=False)
+    body = fields.Html(related='mail_message_id.body', readonly=False)
+    subject = fields.Char(related='mail_message_id.subject', readonly=False)
+
+    email_from_normalized = fields.Char('Normalized From', store=True, compute='_compute_email_from_normalized')
+    mail_group_id = fields.Many2one(
+        'mail.group', string='Group',
+        required=True, ondelete='cascade')
+    mail_message_id = fields.Many2one('mail.message', 'Mail Message', required=True, ondelete='cascade', index=True, copy=False)
+    # Parent and children
+    group_parent_id = fields.Many2one(
```

Comment:
A bit harsh ... shoudln't we compute it simply at message_post level and avoid that kind of costly computation ? Normally peopel should not mess with it, so if we have a single entry point it could be set at that point.

Reply:
But we could imagine one day that we can reply directly from the portal view... Or we can also create message from the backend so we have at least 2 entry points

Reply:
I would just keep the inverse, otherwise the demo data will loose the hierarchy